frontends/election-manager/src/screens/print_test_deck_screen.tsx

```diff
@@ -67,7 +75,41 @@ function PrecinctTallyReport({
   );
 }
 
-interface HandMarkedPaperBallotsParams {
+interface BmdPaperBallotsProps {
+  electionDefinition: ElectionDefinition;
+  onAllRendered: (numBallots: number) => void;
+  precinctId: string;
+}
+
+function BmdPaperBallots({
+  electionDefinition,
+  onAllRendered,
+  precinctId,
+}: BmdPaperBallotsProps): JSX.Element {
+  const { election } = electionDefinition;
+  const ballots = generateTestDeckBallots({ election, precinctId });
```

Comment:
Btw @adghayes I tried moving ballot generation out of these components and to the top-level `PrintTestDeckScreen` as you considered in https://github.com/votingworks/vxsuite/pull/1851#issuecomment-1133058527! With `PrecinctTallyReport`, `BmdPaperBallots`, and `HandMarkedPaperBallots` all taking `ballots` props as inputs

But this interfered with the temporary `React.memo` workaround. Happy to dive into further details in person!

Reply:
Interesting!